docs/intro/tutorial.rst

```diff
@@ -422,6 +433,66 @@ Now crawling dmoz.org yields ``DmozItem`` objects::
          'link': [u'http://www.informit.com/store/product.aspx?isbn=0130211192'],
          'title': [u'XML Processing with Python']}
 
+
+Following links
+===============
+
+Let's say, instead of just scraping the stuff in *Books* and *Resources* pages,
+you want everything that is under the `Python directory
+<http://www.dmoz.org/Computers/Programming/Languages/Python/>`_.
+
+Now that you know how to extract data from a page, why not extract the links
+for the pages you are interested, follow them and then extract the data you
+want for all of them?
+
+Here is a modification to our spider that does just that::
+
+    import scrapy
+    import urlparse
+
+    from tutorial.items import DmozItem
+
+    class DmozSpider(scrapy.Spider):
+        name = "dmoz"
+        allowed_domains = ["dmoz.org"]
+        start_urls = [
+            "http://www.dmoz.org/Computers/Programming/Languages/Python/",
+        ]
+
+        def parse(self, response):
+            for href in response.css("ul.directory.dir-col > li > a::attr('href')"):
+                url = urlparse.urljoin(response.url, href.extract())
+                yield scrapy.Request(url, callback=self.parse_dir_contents)
+
+        def parse_dir_contents(self, response):
+            for sel in response.xpath('//ul/li'):
+                item = DmozItem()
+                item['title'] = sel.xpath('a/text()').extract()
+                item['link'] = sel.xpath('a/@href').extract()
+                item['desc'] = sel.xpath('text()').extract()
+                yield item
+
+Now the `parse()` method only extract the interesting links from the page, builds a
+full absolute URL (since the links can be relative) and yields new requests to
+be sent later, registering as callback the method `parse_dir_contents()` that
+will ultimately scrape the data we want.
+
+What you see here is the Scrapy's mechanism of following links: if you yield a
+Request instead of an Item in a callback method, Scrapy will schedule that
+request to be sent and register a callback method to be executed when that
+request finishes.
```

Comment:
It may worths it to mention a callback method can yield both Items and Requests at the same time, there are no separate "item callbacks" and "request callbacks".